client/client-multi/client-c/src/kaa/kaa_event.c

```diff
@@ -681,15 +681,23 @@ kaa_error_t kaa_event_request_serialize(kaa_event_manager_t *self, size_t reques
 
 static kaa_error_t kaa_event_read_event(kaa_event_manager_t *self, kaa_platform_message_reader_t *reader)
 {
-    KAA_RETURN_IF_NIL2(self, reader, KAA_ERR_BADPARAM);
+
+    kaa_error_t error = KAA_ERR_NONE;
+    char *event_fqn = NULL;
+    uint8_t* event_source = NULL;
+
+    if (!(self) || !(reader)) {
+        error = KAA_ERR_BADPARAM;
+        goto end;
```

Comment:
According to the approved code style we don't use goto.
I know, your suggestion is good, but we have the rule.